cms/djangoapps/contentstore/tests/test_tasks.py

```diff
@@ -207,7 +212,61 @@ def test_register_exams_failure(self, _mock_register_exams_proctoring, _mock_reg
             course_publish.assert_called()
 
 
-class CourseOptimizerTestCase(TestCase):
+class MockCourseLinkCheckTask(Task):
+    def __init__(self):
+        self.status = mock.Mock()
+
+
+class CheckBrokenLinksTaskTest(ModuleStoreTestCase):
+    def setUp(self):
+        super().setUp()
+        self.mock_urls = [
+            ["block-v1:edX+DemoX+Demo_Course+type@vertical+block@1", "http://example.com/valid"],
+            ["block-v1:edX+DemoX+Demo_Course+type@vertical+block@2", "http://example.com/invalid"]
+        ]
+        self.expected_file_contents = [
+            ['block-v1:edX+DemoX+Demo_Course+type@vertical+block@1', 'http://example.com/valid', False],
+            ['block-v1:edX+DemoX+Demo_Course+type@vertical+block@2', 'http://example.com/invalid', False]
+        ]
+
+    @mock.patch('cms.djangoapps.contentstore.tasks.UserTaskArtifact', autospec=True)
+    @mock.patch('cms.djangoapps.contentstore.tasks.UserTaskStatus', autospec=True)
+    @mock.patch('cms.djangoapps.contentstore.tasks._scan_course_for_links')
+    @mock.patch('cms.djangoapps.contentstore.tasks._save_broken_links_file', autospec=True)
+    @mock.patch('cms.djangoapps.contentstore.tasks._write_broken_links_to_file', autospec=True)
+    def test_check_broken_links_stores_broken_and_locked_urls(
+        self,
+        mock_write_links,
+        mock_save_broken_links_file,
+        mock_scan_course_for_links,
+        _mock_user_task_status,
+        mock_user_task_artifact
```

Comment:
Shouldn't these inputs match the mock.patch function names as closely as possible?

Reply:
You're right, that's probably cleaner. I think they all match except for `mock_write_links` so fixing that now.